doc/docs/master/deploy-manage/manage/migrations.md

```diff
@@ -1,285 +1,316 @@
+   You might need to install `jq` and other utilities to run the script.
 
-#### Pausing data loading operations
-**Input repositories** or **input repos** in Pachyderm are repositories created with the `pachctl create repo` command.
-They're designed to be the repos at the top of a directed acyclic graph of pipelines.
-Pipelines have their own output repos associated with them, and are not considered input repos.
-If there are any processes external to pachyderm that put data into input repos using any method
-(the Pachyderm APIs, `pachctl put file`, etc.), 
-they need to be paused.  
-See [Loading data from other sources into pachyderm](../backup_restore/#loading-data-from-other-sources-into-pachyderm) below for design considerations for those processes that will minimize downtime during a restore or migration.
+1. Confirm that each pipeline is started using the `list pipeline` command:
 
-Alternatively, you can use the following commands to stop all data loading into Pachyderm from outside processes.
+   ```bash
+   pachctl list pipeline
+   ```
 
-```
-# Once you have stopped all running pachyderm pipelines, such as with this command,
-# pachctl list pipeline --raw \
-#   | jq -r '.pipeline.name' \
-#   | xargs -P3 -n1 -I{} pachctl stop pipeline {}
+   * If you have switched the ports to stop data loading from outside sources,
+   change the ports back:
 
-# all pipelines in your cluster should be suspended. To stop all
-# data loading processes, we're going to modify the pachd Kubernetes service so that
-# it only accepts traffic on port 30649 (instead of the usual 30650). This way,
-# any background users and services that send requests to your Pachyderm cluster
-# while 'extract' is running will not interfere with the process
-#
-# Backup the Pachyderm service spec, in case you need to restore it quickly
-kubectl get svc/pachd -o json >pach_service_backup_30650.json
+     1. Back up the current configuration:
 
-# Modify the service to accept traffic on port 30649
-# Note that you'll likely also need to modify your cloud provider's firewall
-# rules to allow traffic on this port
-kubectl get svc/pachd -o json | sed 's/30650/30649/g' | kc apply -f -
+        ```bash
+        kubectl get svc/pach -o json >pach_service_backup_30649.json
+        kubectl get svc/etcd -o json >etcd_svc_backup_32379.json
+        kubectl get svc/dash -o json >dash_svc_backup_30080.json
+        ```
 
-# Modify your environment so that *you* can talk to pachd on this new port
-pachctl config update context `pachctl config get active-context` --pachd-address=<cluster ip>:30649
+     1. Modify the services to accept traffic on the corresponding ports:
 
-# Make sure you can talk to pachd (if not, firewall rules are a common culprit)
-pachctl version
-COMPONENT           VERSION
-pachctl             1.9.7
-pachd               1.9.7
-```
+        ```bash
```

Comment:
See my comment to the similar section of _backup_restore_ , where I recommend telling the user what each of these does.

Reply:
added